hydro_lang/src/stream.rs

```diff
@@ -981,6 +981,26 @@ pub(super) fn deserialize_bincode<T: DeserializeOwned>(tagged: Option<syn::Type>
     }
 }
 
+impl<'a, T, C1, C2, B, Order> Stream<(ClusterId<C2>, T), Cluster<'a, C1>, B, Order> {
+    pub fn send_partitioned<Tag, F: Fn((ClusterId<C2>, T)) -> (ClusterId<C2>, T) + 'a>(
```

Comment:
It feels a bit weird to have this as just a utility for `.map` + `.send_bincode_interleaved`, doesn't seem to save _that_ much code. If you'd like to still get this in, what I'd prefer is we create a `trait PartitionStream<T, C2>` with this API and `impl <'a, T, C1, C2> PartitionStream<C2, T> for Stream<(ClusterId<C2>, T), ...`